dashboard/app/models/activity.rb

```diff
@@ -72,4 +72,62 @@ def Activity.recent(limit)
     # yeah, this is a lot like .last -- but I want a dataset not an array
     Activity.order('id desc').limit(limit)
   end
+
+  # Creates a new Activity which will be written eventually to the database. (Note that the id is
+  # nil because it may not have been written yet.) An exception will be thrown if the object does
+  # not pass validation. The object is only written asynchronously if the gatekeeper allows it for
+  # this hostname.
+  def Activity.create_async!(attributes)
+    if Gatekeeper.allows('async_activity_writes', where: {hostname: Socket.gethostname})
```

Comment:
Note that `Socket.gethostname` will not return the 'friendly' hostnames that show up in our SSH prompts, it only gets the ip-based hostnames initially given by the server:

```
ubuntu@frontend-d4:~$ ruby -rsocket -e 'puts Socket.gethostname'
ip-10-113-196-238
```

This might be a bug in how we update the hostname on the server's bootstrap (maybe it needs a reboot or to be updated somewhere else for the running system).